### PR TITLE
Download from correct channel

### PR DESCRIPTION
**Pull request by johnewart**
* Auto-determine the channel to download in the containers based on the channel baked into the CLI
* Perform a HEAD request to see if the local cached download matches the remote file size (fixes corrupt cached files)
* Auto-detect development mode (i.e upload the local binary being executed into the container) without needing to set an environment variable

### CODE REVIEW

File: cli/build.go [hunk 1]
```diff
@@ -704,17 +697,14 @@ func (b BuildData) EnvironmentVariables() []string {
...
+	// Stable version stable URL
+	downloadUrl := "https://bin.equinox.io/c/3fw6ZrH8UFD/yb-stable-linux-amd64.tgz"
 	currentPath, _ := filepath.Abs(".")
 	tmpPath := filepath.Join(currentPath, ".yb-tmp")
-	MkdirAsNeeded(tmpPath)
 	ybPath := filepath.Join(tmpPath, "yb")
-	if _, err := os.Stat(ybPath); err == nil {
-		return ybPath, nil
-	}
+	MkdirAsNeeded(tmpPath)
```
Comment (herbertmd) [inline cli/build.go]: Did you mean to change this to ybPath, since you moved the call to after setting that variable?

File: cli/build.go [hunk 2]
```diff
@@ -431,12 +438,6 @@ func BuildInsideContainer(target BuildTarget, containerOpts BuildContainerOpts,
 		} else {
 			localYbPath = p
 		}
-		// After downloading YB and uploading it to the container, we should delete it locally
```
Comment (beholders-eye) [inline cli/build.go]: Without removing `.yb-tmp/` we'll have to add this to every `.gitignore` that uses the CLI... Why did you remove this?

File: cli/build.go (hunk 1 above)
Comment (beholders-eye) [reply cli/build.go]: Actually tmpPath == path to the temporary directory that will keep the yb binary; ybPath == tmpPath + "/yb", right?

File: cli/build.go [hunk 3]
```diff
@@ -726,8 +716,3 @@ func DownloadYB() (string, error) {
 
 	return ybPath, nil
 }
-
-func DeleteLocalYB() error {
```
Comment (beholders-eye) [inline cli/build.go]: I don't agree with this. Getting yb from the correct channel and auto-detecting devmode don't invalidate the need to cleanup this temporary directory. Maybe we could use a real temporary directory by using `os.TempDir()` (https://godoc.org/os#TempDir), as most modern O.S. clean them up at boot or shutdown.

Comment (johnewart) [reply cli/build.go]: Seems reasonable -- I changed it to use our cache dir, also checked to see if the binary checksum matches the known good one before bothering to try re-downloading

File: cli/build.go (hunk 2 above)
Comment (johnewart) [reply cli/build.go]: Changed, see below / commit comment

File: cli/build.go (hunk 1 above)
Comment (johnewart) [reply cli/build.go]: Ridai is correct -- tmpPath is where we un-archive the binary to, so we need to make it before we explode the archive
